Framework/Algorithms/src/DetectorEfficiencyCor.cpp

```diff
@@ -221,19 +222,24 @@ void DetectorEfficiencyCor::correctForEfficiency(
 
   // Storage for the reciprocal wave vectors that are calculated as the
   // correction proceeds
+  const auto &detectorInfo = m_inputWS->detectorInfo();
   std::vector<double> oneOverWaveVectors(yValues.size());
   for (auto it = dets.cbegin(); it != dets.cend(); ++it) {
-    IDetector_const_sptr det_member =
-        m_inputWS->getInstrument()->getDetector(*it);
+    size_t detIndex;
+    try {
+      detIndex = detectorInfo.indexOf(*it);
+    } catch (std::out_of_range &) {
+      throw Exception::NotFoundError("Detector ID", *it);
+    }
 
+    const auto &det_member = detectorInfo.detector(detIndex);
```

Comment:
This can now be done much simpler (you may need to merge in `master`, the pull request that gives you this functionality was merged yesterday):

```cpp
for(const auto index : spectrumInfo.spectrumDefinition(spectraIn)) {
  const auto detIndex = index.first;
  const auto &det_member = detectorInfo.detector(detIndex);
  // ...
}

Everything starting with `// get a pointer to the detectors that created the spectrum` could then be removed, I think.